docs/ci-services.md

```diff
@@ -0,0 +1,55 @@
+# CI Services
+
+The CI service blob defines the supported CI services the user will be
+presented with when running the scaffolder.
+
+## Example
+
+```javascript
+{
+  TravisCI: {scaffolder: travisScaffolder, public: true}
+}
+```
+
+## Config
+
+### keys: __string__
+
+Keys in the CI services blob define the options the user will be presented with
+when choosing their desired CI service.
+
+example: `TravisCI`
+
+### values: __object__
+
+Values in the CI services blob define the actual mechanisms and configuration
+that will be used by the JavaScript Scaffolder to scaffold the CI service for
+the user's project.
+
+#### scaffolder: __function__ (_required_)
+
+Consumer-defined plugin(??) called when this ci service is selected by the
+user in the CLI. Will be called with the following arguments:
+
+* `projectRoot` __string__ full path to directory where scaffolder was initiated
+* `vcs` __object__
+  * `host` __string__ VCS hosting service to generate the repo with (e.g., `github`)
+  * `owner` __string__ account name on the host service for the repository to
+    generate (e.g., `form8ion`)
+  * `name` __string__ repository name to generate (e.g., `javascript-scaffolder`)
+* `visibility` __string__ visibility of the project to be scaffolded (e.g., `Public`, `Private`)
```

Comment:
do we validate that `Public` and `Private` are the valid options here with joi? if we limit to only those two, it would probably be better to say something like "valid values are: " instead of "e.g." in that case